ssh/ssh.go

```diff
@@ -121,7 +125,7 @@ func getControlDir(osEnv config.Environment) (string, error) {
 		// os.ErrExist, but that's not available on Go 1.11.
 		perr, ok := err.(*os.PathError)
 		if !ok || perr.Err != syscall.EEXIST {
-			return ioutil.TempDir("", "sock-*")
+			return os.MkdirTemp("", "sock-*")
```

Comment:
I think we still have a problem here.  If we set `dir` to `/tmp`, we then try to create `/tmp/git-lfs`, and if it fails, we use `$TMPDIR` to create a random path, which is still going to be too long.  That means the second time the same user runs Git LFS, or the first time a different user runs Git LFS, `/tmp/git-lfs` will fail to be created and we'll be back to fix this all over again.

Reply:
In that case, I think we should bubble up the error instead of suppressing it.

On another note, what do you think of using the user's cache directory for storing the control socket? We could use `os.UserCacheDir()`

Reply:
I've updated the patch to create a random `sock-*` dir under `/tmp` on darwin using `os.MkdirTemp`